spi/control-plane/control-plane-spi/src/main/java/org/eclipse/edc/connector/spi/policydefinition/PolicyDefinitionService.java

```diff
@@ -64,4 +64,15 @@ public interface PolicyDefinitionService {
 
     @NotNull
     ServiceResult<PolicyDefinition> create(PolicyDefinition policy);
+
+    /**
+     * Updates a policy. If the policy does not yet exist, {@link ServiceResult#notFound(String)} will be returned.
+     *
+     * @param policyId the ID of the policy to update
+     * @param policy the contents of the policy.
+     *
+     * @return successful if updated, a failure otherwise.
+     */
+    @NotNull
+    ServiceResult<Void> update(String policyId, PolicyDefinition policy);
```

Comment:
Why is the id explicitly passed here?

Reply:
That's correct, we can only use PolicyDefinition.
But, one of the review I saw in #PR [2156](https://github.com/eclipse-edc/Connector/pull/2516/files) states that `ServiceImpl` should not depend upon controllers, so i tried to follow the same.

Please let me know, if I shall change it or maintain the consistent behaviour of `update` semantics in the code base

Thank you.

Reply:
I don't understand why not passing the policy id explicitly and using the id from the policy definition is related to a service implementation depending on a controller.

Reply:
there is issue #2531 which takes care of that. can we resolve this then?